src/board.py

```diff
@@ -138,18 +117,18 @@ def __getitem__(self, square: Square | Indices | None) -> Piece | None:
             The piece on the given square.
         """
         if square:
-            i, j = self._indices(square) if isinstance(square, Square) else square
-            return self._board[i][j]
+            square = Square(square)
+            return self._board[square.rank][square.file]
 
-    def __delitem__(self, square: Square | Indices | None):
+    def __delitem__(self, square: Square | str | None):
```

Comment:
It is indeed a redundancy. However, I consider this a safety option more than a useful option. If anyone wants to fork our engine and uses `del board[square]` they will damage the board, there is zero usage for the actual `del` operation, so we kind of protect the user of this class, right?

Reply:
Indeed, here we don't delete the element, otherwise we would have a board with gaps. Semantically speaking, I think it is fine to do it that way, it is consistent with how a board works. The fact that it doesn't delete the element of the list is implementation detail, after all `self._board` isn't meant to be accessed directly and the fact we implemented it as a list of lists is implementation detail. For example we could implement a dict with keys the positions in algebraic notation that aren't empty and then we would call del on the dict instead of setting the list position with None as we do now but that's implementation detail.